### PR TITLE
fix(react-storybook-addon): transform decorator to function in withAriaLive()

### LINKED ISSUE #32010: [Bug]: motion example is flickering in docs
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://react.fluentui.dev/?path=/docs/utilities-motion-createmotioncomponent--default

## Actual Behavior
- Open https://react.fluentui.dev/?path=/docs/utilities-motion-createmotioncomponent--default
- Notice that animation is flickering (too fast)
  

## Expected Behavior

- Animation does not have flickering as it has `playbackRate = 0.2`

### Logs

```shell
N/A
```

### Requested priority

High

### Products/sites affected

Fluent docs

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by layershifter**
## Previous Behavior

#31794 added a new decorator `withAriaLive()`. That decorator has conditional rendering:

https://github.com/microsoft/fluentui/blob/1c2c2003bff234ce6a4a043453479889c153a4b1/packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx#L16

The problem is that it's implemented according to Storybook 6 docs and stories as a function:

https://github.com/microsoft/fluentui/blob/1c2c2003bff234ce6a4a043453479889c153a4b1/packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx#L6

That causes an issue with a conditional rendering due effect order:

```
- useEffect: in story (as `withAriaLive()` call a function) [RENDER 1]
- useEffect: AriaLiveWrapper (mount) [RENDER 1]
```

This issue breaks stories that rely on `useEffect(fn, [])` as they are executed in wrong order.

## New Behavior

~The PR changes decorators to be called as React components. The same is suggested in Storybook 7 & 8 docs (https://storybook.js.org/docs/writing-stories/decorators#component-decorators).~ The PR updates `withAriaLive()` with an explicit condition for VR tests as VR tooling will fail to get stories for steps otherwise (see `findSteps()` in `node_modules/storywright/src/StoryWrightProcessor/GetStories.js`) 

This fixed the order or effects:

```
- useEffect: AriaLiveWrapper (mount) [RENDER 1]
- useEffect: in story [RENDER 2]
```

## Related Issue(s)

Fixes #32010.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx [hunk 1]
```diff
@@ -2,16 +2,22 @@ import * as React from 'react';
```
Comment (fabricteam) [inline packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx:2]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32011&commitId=31ef8656c3ae1317fcfc9985e12a1d4f1696e49d&env=prod&clientType=FLUENTUI)

_Diffs exceeded the limit allowed for comments. Please use the report link above to check your diffs._


<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by Hotell** [comment]
can we postpone this until https://github.com/microsoft/fluentui/pull/31939 lands as there seems to be a lot of VR diffs caused by this change ?

File: packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx [hunk 2]
```diff
@@ -1,17 +1,34 @@
...
+export const withAriaLive = (Story: () => JSX.Element, context: FluentStoryContext) => {
+  const { parameters } = context;
+  const { mode } = parameters;
+  const isVrTest = mode === 'vr-test';
 
-export const withAriaLive = (StoryFn: () => JSX.Element) => {
-  return <AriaLiveWrapper>{StoryFn()}</AriaLiveWrapper>;
+  // - VR tests don't need the AriaLiveAnnouncer
+  // - VR tests tooling supports only functions as story renderers :(
+  if (isVrTest) {
```
Comment (Hotell) [inline packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx:13]: i'm wondering if having ability to actually configure on addon registration level which decorators should be loaded would be a more robust solution then having this internal "hack" per say.

if we register decorators via storybook function name of decorator would be exposed which could be then used.

https://storybook.js.org/docs/6/addons/addons-api#makedecorator-api

WDYT ?

Comment (layershifter) [reply packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx:13]: Not sure that I follow, @Hotell can you please expand with a code example?

Comment (Hotell) [reply packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx:13]: sure. (NOTE: that makeDecoratorApi is not needed - I checked )

**Option 1:**

image


**Options 2:**

similar to yours but explicit public API  from addon POV

image

Comment (layershifter) [reply packages/react-components/react-storybook-addon/src/decorators/withAriaLive.tsx]: @Hotell I implemented option 2 👍

**Comment by layershifter** [comment]
> can we postpone this until #31939 lands as there seems to be a lot of VR diffs caused by this change ?

@Hotell It was a bug in my code, now there are no regressions:

image

File: packages/react-components/react-storybook-addon/src/decorators/withFluentProvider.tsx [hunk 3]
```diff
@@ -23,8 +23,14 @@ const getActiveFluentTheme = (globals: FluentGlobals) => {
 export const withFluentProvider = (StoryFn: () => JSX.Element, context: FluentStoryContext) => {
   const { globals, parameters } = context;
   const { mode } = parameters;
-  const isVrTest = mode === 'vr-test';
 
+  const shouldDisable = parameters.reactStorybookAddon?.disabledDecorators?.includes('FluentProvider');
+
+  if (shouldDisable) {
+    return StoryFn();
+  }
```
Comment (Hotell) [inline packages/react-components/react-storybook-addon/src/decorators/withFluentProvider.tsx:31]: **nit:** can we make this more dry in all decorators (within this addon only ofc) ?

something like:

```suggestion
  if (isDecoratorDisabled(parameters,'FluentProvider')) {
    return StoryFn();
  }
```

File: apps/vr-tests-react-components/.storybook/preview.js [hunk 4]
```diff
@@ -69,4 +69,10 @@ setAddon({
 });
 
 /** @type {import("@fluentui/react-storybook-addon").FluentParameters} */
-export const parameters = { layout: 'none', mode: 'vr-test' };
+export const parameters = {
```
Comment (Hotell) [inline apps/vr-tests-react-components/.storybook/preview.js:72]: follow-up https://github.com/microsoft/fluentui/issues/32054

**Review by Hotell — APPROVED** [review]
added some suggestion but overall

LGTM

ty

File: packages/react-components/react-storybook-addon/src/decorators/withFluentProvider.tsx (hunk 3 above)
Comment (layershifter) [reply packages/react-components/react-storybook-addon/src/decorators/withFluentProvider.tsx]: Applied 👍

**Pull request merged**